googleapiclient/http.py

```diff
@@ -1202,7 +1202,17 @@ def _serialize_request(self, request):
 
     if request.http is not None and hasattr(request.http.request,
         'credentials'):
-      request.http.request.credentials.apply(headers)
+      credentials = request.http.request.credentials
```

Comment:
I've played around with a couple different implementations here and the tests have led me to recognize a few other corner cases and/or places where this isn't ideal.  Specifically, 

* `Credentials` doesn't have an `access_token` property.
* `OAuth2Credentials` does have an `access_token` property.
* `AssertionCredentials` (which eventually subclass `OAuth2Credentials` _do_ have an `access_token` property, but it gets set to `None` in `__init__` which leaves `AssertionCredentials` with a non-working `apply` method
  - `apply` looks to be inherited from `OAuth2Credentials` which assumes that `access_token` is a string

I'm not super familiar with the architecture of this library and the architecture of the `oauth2client` library and how much their requirements are supposed to interoperate.  Having looked at this more, it feels like the bug might actually be in `oauth2client.client.AssertionCredentials.apply`.  If that method were to become a no-op (at least if `self.access_token is None`), then the problems _here_ would go away.  I'm not sure if that's the right thing to do _there_ though.

Maybe this isn't the right place to raise that issue (but it looks like a lot of the committers to both libraries are the same) -- and I suppose the fix here would still need to be in place to handle interoperability with past versions of that library.  Anyway, any advice you can give on this front is welcome :-)